Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/common/taimaba/TaimabaApi.java

```diff
@@ -0,0 +1,340 @@
+                    reader.beginArray();
+
+                    while (reader.hasNext()) {
+                        PostImage postImage = readPostImage(reader, builder, endpoints);
+                        if (postImage != null) {
+                            files.add(postImage);
+                        }
+                    }
+
+                    reader.endArray();
+                    break;
+                default:
+                    // Unknown/ignored key
+                    reader.skipValue();
+                    break;
+            }
+        }
+        reader.endObject();
+
+        // The file from between the other values.
+        if (fileName != null && fileExt != null) {
+            Map<String, String> args = makeArgument("tim", fileName,
+                    "ext", fileExt);
+            PostImage image = new PostImage.Builder()
+                    .thumbnailUrl(endpoints.thumbnailUrl(builder, false, args))
+                    .spoilerThumbnailUrl(endpoints.thumbnailUrl(builder, true, args))
+                    .imageUrl(endpoints.imageUrl(builder, args))
+                    .filename(Parser.unescapeEntities(fileName, false))
+                    .extension(fileExt)
+                    .imageWidth(fileWidth)
+                    .imageHeight(fileHeight)
+                    .spoiler(fileSpoiler)
+                    .size(fileSize)
+                    .build();
+            // Insert it at the beginning.
+            files.add(0, image);
+        }
+
+        builder.images(files);
+
+        if (builder.op) {
+            // Update OP fields later on the main thread
+            Post.Builder op = new Post.Builder();
+            op.closed(builder.closed);
+            op.archived(builder.archived);
+            op.sticky(builder.sticky);
+            op.replies(builder.replies);
+            op.images(builder.imagesCount);
+            op.uniqueIps(builder.uniqueIps);
+            op.lastModified(builder.lastModified);
+            queue.setOp(op);
+        }
+
+        Post cached = queue.getCachedPost(builder.id);
+        if (cached != null) {
+            // Id is known, use the cached post object.
+            queue.addForReuse(cached);
+            return;
+        }
+
+        if (countryCode != null && countryName != null) {
+            HttpUrl countryUrl = endpoints.icon("country",
+                    makeArgument("country_code", countryCode));
+            builder.addHttpIcon(new PostHttpIcon(countryUrl, countryName + "/" + countryCode));
+        }
+
+        if (trollCountryCode != null && countryName != null) {
+            HttpUrl countryUrl = endpoints.icon("troll_country",
+                    makeArgument("troll_country_code", trollCountryCode));
+            builder.addHttpIcon(new PostHttpIcon(countryUrl, countryName + "/t_" + trollCountryCode));
+        }
+
+        queue.addForParse(builder);
+    }
+
+    private PostImage readPostImage(JsonReader reader, Post.Builder builder,
```

Comment:
my idea is using a placeholder thumbnail so you can still download the files.
tested and it works, (currently just using a imagehoster website) altough it will display an error "failed to show image, deep-zooming loading failed"on tapping on it as obviously swf format isn't natively supported on android and it tries to load that.

but i think we could go on from there. there's no static image on 420chan unfortunately as there simply is nothing displayed but a link to the file itself, however fully disabling image display would essentially render the whole board pretty much useless imho.

i'll commit my current solution for further review.

Reply:
whatever is to be done has to be done for mü3 files as well

Reply:
i think doing something similar to 462029b is the way to go
I've tested that on my private fork and it's working.